src/main.js

```diff
@@ -1599,7 +1599,11 @@ const DiscoveryFeedApplication = new Lang.Class({
     Extends: Gtk.Application,
 
     _init: function() {
-        this.parent({ application_id: pkg.name });
+        this.parent({
+            application_id: pkg.name,
+            inactivity_timeout: 12000,
+            flags: Gio.ApplicationFlags.IS_SERVICE
+        });
```

Comment:
I think you can avoid this if you pass `--gapplication-service` to the `Exec` line in the dbus service file.